addons/website/static/src/snippets/s_table_of_content/options.js

```diff
@@ -0,0 +1,152 @@
+        }
+    },
+    /**
+     * Change the navbar position.
+     *
+     * @see this.selectClass for parameters
+     */
+    navbarPosition: function (previewMode, widgetValue, params) {
+        const $navbar = this.$target.find('.s_table_of_content_navbar_wrap');
+        const $mainContent = this.$target.find('.s_table_of_content_main');
+        if (widgetValue === 'top' || widgetValue === 'left') {
+            $navbar.prev().before($navbar);
+        }
+        if (widgetValue === 'left' || widgetValue === 'right') {
+            $navbar.removeClass('s_table_of_content_horizontal_navbar col-lg-12').addClass('s_table_of_content_vertical_navbar col-lg-3');
+            $mainContent.removeClass('col-lg-12').addClass('col-lg-9');
+            $navbar.find('.s_table_of_content_navbar').removeClass('list-group-horizontal-md');
+        }
+        if (widgetValue === 'right') {
+            $navbar.next().after($navbar);
+        }
+        if (widgetValue === 'top') {
+            $navbar.removeClass('s_table_of_content_vertical_navbar col-lg-3').addClass('s_table_of_content_horizontal_navbar col-lg-12');
+            $navbar.find('.s_table_of_content_navbar').addClass('list-group-horizontal-md');
+            $mainContent.removeClass('col-lg-9').addClass('col-lg-12');
+        }
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * @private
+     */
+    _generateNav: function (ev) {
+        let $nav = this.$target.find('.s_table_of_content_navbar');
+        const $headings = this.$target.find(this.targetedElements);
+        $nav.empty();
+        _.each($headings, el => {
+            const $el = $(el);
+            const id = _.uniqueId('table_of_content_heading_');
+            $('<a>').attr('href', "#" + id)
+                    .addClass('list-group-item list-group-item-action py-2 border-0')
+                    .text($el.text())
+                    .appendTo($nav);
+            $el.attr('id', id);
+            $el[0].dataset.anchor = this.IsAnimateScrolling === 'true' ? 'true' : '0';
+        });
+        $nav.find('a:first').addClass('active');
+    },
+    /**
+     * Generate the navbar if the content changed.
+     * 
+     * @private
+     */
+    _observeContentChange: function (ev) {
+        const self = this;
+        const targetNode = this.$target.find('.s_table_of_content_main')[0];
+        const config = {attributes: false, childList: true, subtree: true, characterData: true};
+        this.observer = new MutationObserver(function (mutationsList) {
+            _.each(mutationsList, el => {
+                self._generateNav();
+            });
+        });
+        this.observer.observe(targetNode, config);
+    },
+    /**
+     * @override
+     */
+    _computeWidgetState: function (methodName, params) {
+        let $headings = '';
+        let $navbar = '';
```

Comment:
Why defining this here, define that where needed and wrap the case clauses in an instructions block.

```javascript
case XXX: {
    const ...
    break;
}
```